signature/cose/envelope.go

```diff
@@ -439,7 +436,9 @@ func generateProtectedHeaders(req *signature.SignRequest, protected cose.Protect
 // during Sign process.
 func generateUnprotectedHeaders(req *signature.SignRequest, signer signer, unprotected cose.UnprotectedHeader) {
 	// signing agent
-	unprotected[headerLabelSigningAgent] = req.SigningAgent
+	if strings.TrimSpace(req.SigningAgent) != "" {
```

Comment:
nit: we don't need to check space here since it is not a user-typed input.
```suggestion
	if req.SigningAgent != "" {
```